### PR TITLE
PLAT-980 | Add filters by classification on Content Classifier

### PR DESCRIPTION
**Pull request by kvas-damian**

### CODE REVIEW

File: app/Http/Controllers/Api/Filters/TaxonomyTermsFilter.php [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<?php namespace App\Http\Controllers\Api\Filters;
+
+
+class TaxonomyTermsFilter extends ApiFilter
```
Comment (bognix) [inline app/Http/Controllers/Api/Filters/TaxonomyTermsFilter.php]: how about simple unit test for this method?

File: resources/assets/js/admin/components/contentClassifier/ContentClassifier.vue [hunk 2]
```diff
@@ -1,50 +1,76 @@
...
+				>
+					<a>Po id</a>
+				</li>
+			</ul>
+		</div>
+
+		<wnl-content-classifier-filter-by-classification
+			v-show="activeTab === TABS.BY_CLASSIFICATION"
+			:content-types="contentTypes"
+			@search="onByTagSearch"
```
Comment (bognix) [inline resources/assets/js/admin/components/contentClassifier/ContentClassifier.vue]: one method is called `onByTagSearch` and the second one is `onSearchById` - could you unify the placement of `search` word?

File: resources/assets/js/admin/components/contentClassifier/ContentClassifier.vue [hunk 3]
```diff
@@ -237,16 +270,19 @@ export default {
...
 			} else {
-				this.selectedItemIds.splice(index, 1);
+				this.selectedItems.splice(index, 1);
 			}
 		},
 		selectAll() {
-			this.selectedItemIds = this.filteredContent.map(item => item.id);
+			this.selectedItems = [...this.filteredContent];
+		},
+		getselectedCountsByContentType(contentType) {
```
Comment (bognix) [inline resources/assets/js/admin/components/contentClassifier/ContentClassifier.vue]: `getSelectedCountsByContentType`

File: resources/assets/js/admin/components/contentClassifier/ContentClassifierFilterByClassification.vue [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+					placeholder="Zacznij pisać aby wyszukać tag"
+					@change="onTagSelect"
+				/>
+			</div>
+		</div>
+
+		<div class="content-classifier-filter__active-filters">
+			<h5 class="title is-5">Aktywne filtry</h5>
+
+			<div class="content-classifier-filter__active-filters__list margin bottom" v-if="byTaxonomyTermsFilter.length || byTagsFilter.length">
```
Comment (bognix) [inline resources/assets/js/admin/components/contentClassifier/ContentClassifierFilterByClassification.vue]: the part you have in `v-if` I would move to computed variable. You use it in `submitDisabled` as well.

File: resources/assets/js/admin/components/contentClassifier/ContentClassifierFilterByClassification.vue [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+		WnlTaxonomyTermWithAncestors,
+	},
+	props: {
+		contentTypes: {
+			type: Object,
+			required: true,
+		}
+	},
+	data() {
+		const isActiveContentTypes = Object.keys(this.contentTypes).reduce(
```
Comment (bognix) [inline resources/assets/js/admin/components/contentClassifier/ContentClassifierFilterByClassification.vue]: `activeContentTypesMap` but not sure if it's better.
For me the `is` prefix was confusing.

File: resources/assets/js/admin/components/contentClassifier/ContentClassifierFilterByClassification.vue [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+				this.byTagsFilter.push(tag);
+			}
+		},
+		onTermSelect(term) {
+			if (!this.byTaxonomyTermsFilter.find(({id}) => id === term.id)) {
+				this.byTaxonomyTermsFilter.push(term);
+			}
+		},
+		onTagDelete(tag) {
+			const index = this.byTagsFilter.findIndex(({id}) => id === tag.id);
```
Comment (bognix) [inline resources/assets/js/admin/components/contentClassifier/ContentClassifierFilterByClassification.vue]: check if index is not `-1`

File: resources/assets/js/admin/components/contentClassifier/ContentClassifier.vue [hunk 7]
```diff
@@ -1,50 +1,76 @@
...
+
+		<wnl-content-classifier-filter-by-classification
+			v-show="activeTab === TABS.BY_CLASSIFICATION"
+			:content-types="contentTypes"
+			@search="onByTagSearch"
+		/>
+
+		<wnl-content-classifier-filter-by-ids
+			v-show="activeTab === TABS.BY_ID"
+			:contentTypes="contentTypes"
```
Comment (bognix) [inline resources/assets/js/admin/components/contentClassifier/ContentClassifier.vue]: `:content-types`

File: resources/assets/js/admin/components/contentClassifier/ContentClassifier.vue [hunk 8]
```diff
@@ -199,11 +220,23 @@ export default {
 				return item;
 			});
 		},
-		async onSearch() {
-			this.isLoading = true;
-			this.selectedItemIds = [];
+		async onSearchById(filters) {
```
Comment (bognix) [inline resources/assets/js/admin/components/contentClassifier/ContentClassifier.vue]: I was thinking about moving the logic from `onSearchById` and `onByTagSearch` to the dedicated components. On the other hand that would mean you need to pass the `fetch` method to those components which mean one prop more.

It's logic encapsulation vs props count debate.

File: resources/assets/js/admin/components/slides/annotations/AnnotationsList.vue [hunk 9]
```diff
@@ -15,13 +15,11 @@
...
-					<span
-						class="tag"
+					<wnl-tag
 						v-for="tag in annotation.tags"
-						:key="tag.id"
-						:style="{backgroundColor: getColourForStr(tag.name)}">
-						{{tag.name}}
-					</span>
+					 :key="tag.id"
+						:tag="tag"
```
Comment (bognix) [inline resources/assets/js/admin/components/slides/annotations/AnnotationsList.vue]: indent.

File: resources/assets/js/admin/components/contentClassifier/ContentClassifier.vue (hunk 8 above)
Comment (kvas-damian) [reply resources/assets/js/admin/components/contentClassifier/ContentClassifier.vue]: What was the outcome of the thinking process? :trollface:

File: resources/assets/js/admin/components/taxonomies/TaxonomyTermEditorForm.vue [hunk 10]
```diff
@@ -0,0 +1,139 @@
+<template>
```
Comment (rogatty) [inline resources/assets/js/admin/components/taxonomies/TaxonomyTermEditorForm.vue]: @kvas-damian this file was deleted in https://github.com/bethinkpl/wnl-platform/pull/1293/files#diff-2524b0995d385ab87dcf568a8d1c36b9

**Pull request merged**